### DISCUSSION TITLE
Miller 6.0.0 beta release

### DISCUSSION THREAD

**Discussion by johnkerl** [johnkerl/miller#759] [Announcements]
This is a beta release for the upcoming 6.0.0 release of [Miller](https://miller.readthedocs.io/en/latest/).

Miller is like `awk`, `sed`, `cut`, `join`, and `sort` for name-indexed data such as CSV, TSV, and tabular JSON.

Miller 6 is a major, exciting update with lots of features, documentation improvements, full Windows support, and more.

_This is superseded by the Miller 6 release: https://github.com/johnkerl/miller/discussions/851_

# Status

This is marked as a pre-release -- you can get the binaries (for Linux, Mac, and Windows) by downloading them from this release page. Meanwhile tools like `brew`, `apt`, `chocolatey`, etc will still give you Miller 5 until the _official_ Miller 6.0.0 release which is forthcoming.

# Features

Release notes can be found at https://miller.readthedocs.io/en/latest/new-in-miller-6. Please also see full documentation at https://miller.readthedocs.io/en/latest.

# Goals for the beta

The beta release is feature-complete, while final performance optimizations are in progress -- you may notice slower processing times for the beta. The 6.0.0 version will not be released until these performance optimizations are complete.

Please comment on this page, or file an issue at https://github.com/johnkerl/miller/issues, with any and all feedback, criticism, comments, etc.

<hr /><em>This discussion was created from the release <a href='https://github.com/johnkerl/miller/releases/tag/v6.0.0-beta'>Miller 6.0.0 beta release</a>.</em>

**Comment by aborruso** [comment]
Dear @johnkerl only to say thank you very much for this great release, you put a lot of care, ideas and quality into it. 
Miller has become a daily companion for me, it is a real pleasure to see it grow.

🙏Thank you so much

**Comment by johnkerl** [comment]
New binaries attached to [the pre-release](https://github.com/johnkerl/miller/releases/tag/v6.0.0-beta) today incorporate [#765](https://github.com/johnkerl/miller/pull/765) which is a 40% reduction in runtime for large files. Two more performance PRs are in prep.

**Comment by johnkerl** [comment]
New binaries attached to [the pre-release](https://github.com/johnkerl/miller/releases/tag/v6.0.0-beta) today incorporate several recent performance-related PRs -- see [#786](https://github.com/johnkerl/miller/pull/786) for details.

**Comment by johnkerl** [comment]
New binaries attached to [the pre-release](https://github.com/johnkerl/miller/releases/tag/v6.0.0-beta) today incorporate all currently known release-blocking issues. The `mlr version` output now shows `6.0.0-rc` to indicate this is a release candidate. I hope to release this very soon, barring any new feedback.
